client/components/UserMenu.tsx

```diff
@@ -79,6 +86,30 @@ export default function UserMenu() {
                     <Editor />
                 </DialogContent>
             </Dialog>
+
+            <Dialog open={announcementsOpen} onOpenChange={setAnnouncementsOpen}>
+                <DialogContent className="min-w-[600px] max-w-[40vw]">
+                    <DialogHeader>
+                        <DialogTitle className="flex justify-center">Announcements</DialogTitle>
+                    </DialogHeader>
+                    <div className="max-h-[50vh] space-y-4 overflow-y-auto">
+                        {announcementList.length > 0 ? (
+                            announcementList.map((announcement, index) => (
+                                <div key={index} className="rounded-lg border bg-muted p-3">
+                                    <p className="text-sm text-muted-foreground">
+                                        {new Date(announcement.time).toLocaleString()}
+                                    </p>
+                                    <p>{announcement.message}</p>
+                                </div>
+                            ))
+                        ) : (
+                            <p className="text-center text-muted-foreground">
+                                No announcements available.
+                            </p>
+                        )}
+                    </div>
+                </DialogContent>
+            </Dialog>
```

Comment:
Consider swapping this out for a [popover](https://ui.shadcn.com/docs/components/popover) instead.